### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by AdamBuijks42**
assignment and unassignment issues solved

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -435,6 +435,9 @@ static void events_cb(void *arg)
         g_tap_tempo[actuator_id].time = 0;
         g_tap_tempo[actuator_id].max = 0;
         g_tap_tempo[actuator_id].state = TT_INIT;
+
+        //clear assignment mode
+        g_current_assignment[actuator_id]->mode = CC_MODE_MOMENTARY;
```
Comment (Janjhl) [inline src/main.c]: What is the purpose of this statement? it's not clearing but setting momentary mode

File: src/main.c [hunk 2]
```diff
@@ -447,10 +450,25 @@ static void events_cb(void *arg)
...
-        assignment->value = set_value->value;
-        update_leds(assignment);
-        update_lcds(assignment);
+        cc_assignment_t *assignment = cc_assignment_get(set_value->assignment_id);
+
+        if (assignment->mode & CC_MODE_OPTIONS)
+        {
+            assignment->list_index = set_value->value;
+
+            hw_led_set(assignment->actuator_id, LED_R, LED_OFF, 0, 0);
```
Comment (Janjhl) [inline src/main.c]: Please handle this properly in update_leds if this is needed

**Review by Janjhl — CHANGES_REQUESTED** [review]
Some minor oddities, functionally ok

File: src/main.c (hunk 1 above)
Comment (AdamBuijks42) [reply src/main.c]: I thought this way I would set the actuator to its init state

**Pull request merged**